### PR TITLE
Flag to include transactions from children

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 1]
```diff
@@ -109,46 +114,71 @@ const TransactionsCollectionQuery = {
...
       ) {
-        const incognitoProfile = await req.remoteUser.getIncognitoProfile();
-        if (incognitoProfile) {
-          fromCollectiveCondition = { [Op.or]: [fromAccount.id, incognitoProfile.id] };
+        const fromAccountUser = await fromAccount.getUser();
+        if (fromAccountUser) {
+          const incognitoProfile = await fromAccountUser.getIncognitoProfile();
+          if (incognitoProfile) {
+            fromAccountCondition.push(incognitoProfile.id);
+          }
```
Comment (znarf) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:137]: Refactoring opportunity to fix what seems a wrong implementation of incognito filter.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 2]
```diff
@@ -109,46 +114,71 @@ const TransactionsCollectionQuery = {
...
-        if (incognitoProfile) {
-          accountConditions.push({ CollectiveId: incognitoProfile.id });
+        const accountUser = await account.getUser();
+        if (accountUser) {
+          const incognitoProfile = await accountUser.getIncognitoProfile();
+          if (incognitoProfile) {
+            accountCondition.push(incognitoProfile.id);
+          }
         }
       }
```
Comment (znarf) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:171]: Refactoring opportunity to fix what seems a wrong implementation of incognito filter.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 3]
```diff
@@ -206,8 +236,8 @@ const TransactionsCollectionQuery = {
     if (args.hasExpense !== undefined) {
       where.push({ ExpenseId: { [args.hasExpense ? Op.ne : Op.eq]: null } });
     }
-    if (args.hasOrder) {
-      where.push({ OrderId: { [Op.ne]: null } });
+    if (args.hasOrder !== undefined) {
+      where.push({ OrderId: { [args.hasOrder ? Op.ne : Op.eq]: null } });
```
Comment (znarf) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:240]: Not related but align code style with hasExpense above.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 4]
```diff
@@ -109,46 +114,71 @@ const TransactionsCollectionQuery = {
...
+          }
         }
       }
 
       where.push({
         [Op.or]: [
           { UsingGiftCardFromCollectiveId: fromAccount.id, type: 'CREDIT' },
-          { FromCollectiveId: fromCollectiveCondition },
+          // prettier, please keep line break for readability please
+          { FromCollectiveId: fromAccountCondition },
```
Comment (kewitz) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:145]:
```suggestion
          { FromCollectiveId: { [Op.in]: fromAccountCondition } },
```

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 5]
```diff
@@ -109,46 +114,71 @@ const TransactionsCollectionQuery = {
...
+          }
         }
       }
 
-      where.push({ [Op.or]: accountConditions });
+      where.push({
+        [Op.or]: [
+          { UsingGiftCardFromCollectiveId: account.id, type: 'DEBIT' },
+          // prettier, please keep line break for readability please
+          { CollectiveId: accountCondition },
```
Comment (kewitz) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:177]:
```suggestion
          { CollectiveId: { [Op.in]: accountCondition } },
```

**Review by kewitz — COMMENTED** [review]
The code looks good, we just need to include the conditional for the array of ids.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts (hunk 4 above)
Comment (znarf) [reply server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:145]: @kewitz `Op.in` is implicit if you pass an array?

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts (hunk 1 above)
Comment (znarf) [reply server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:137]: Previous implementation was actually "ok" given conditions to enter the block.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts (hunk 2 above)
Comment (znarf) [reply server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:171]: Previous implementation was actually "ok" given conditions to enter the block.

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts (hunk 4 above)
Comment (znarf) [reply server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:145]: As highlighted there:
https://www.bennadel.com/blog/3302-you-can-use-arrays-in-field-equality-checks-within-a-sequelize-where-clause-in-node-js.htm

**Pull request merged**